src/lib/shims.js

```diff
@@ -239,13 +257,22 @@ class Shim {
   }
 
   async _registerContentScripts() {
+    // TODO: remove this workaround for Bug 1851173 once webcompat
+    // isn't going to be hitting it anymore.
+    await promiseScriptingAPIInitialized;
     if (
       this.contentScripts.length &&
       !this._contentScriptRegistrations.length
     ) {
       const matches = [];
       for (const options of this.contentScripts) {
         matches.push(options.matches);
+        // NOTE: porting this call to scripting.registerContentScripts
+        // would require Bug 1764572 to have been fixed in all Gecko
+        // version where we would need to run the webcompat builtin,
+        // otherwise the Firebase Shim would not be able to be registered
```

Comment:
And actually, in hindsight this might be a good idea anyway, since if users disable serviceWorkers, this can keep the site working a bit better for them regardless of whether they're in PBM or not.

Reply:
If we can live without cookieStoreIds, then yeah we could port Shim to the scripting API and avoid the workaround added to avoid hitting [Bug 1851173](https://bugzilla.mozilla.org/1851173) (and the big inline comment added to explain why that workaround is there).

The gain for Shim may not be as the one for Injections, because it seems that the Shim may also depend from the background page covering other parts of the expected behavior for the shim (e.g. the parts covered by the webRequest listeners), and so even if the content script would still be injected while the background page may not be around (which would be the case if there are so many crashes that we would stop to reload the background page automatically and leave the user to choose if they want to try again) the shim may not still work if it also depends from what the background page was meant to cover.

@wisniewskit based on your deeper knowledge about what the shims are expected to be doing and how they collaborate with the background page to achieve that, do you think that there may be any chance that a shim content script without the background page running could misbehave?
would you mind to point me to a couple of cases that you think would be worth for me to test out explicitly and look into more deeply to get a better picture of this part and form a more informed opinion on my side about what we would expect to happen (vs. what should happen)?

Reply:
As a side note, looking to the firebase.js shim, that one should still work fine without the background page, and so as I was expecting some shim do not depend from the background page still being around (and I could look for specific properties in the list of shims to determine which ones depends on the background page, e.g. I suspect that the ones with `needsShimHolders` or the ones redirecting urls may be part of the shims that depends also on the background page being running)